proxy/http/HttpSM.cc

```diff
@@ -2717,7 +2717,7 @@ HttpSM::tunnel_handler_post_or_put(HttpTunnelProducer *p)
     tunnel.reset();
     // When the ua completed sending it's data we must have
     //  removed it from the tunnel
-    ink_release_assert(ua_entry->in_tunnel == false);
+    ua_entry->in_tunnel     = false;
```

Comment:
Is this necessary?

Reply:
May no longer be necessary.  I ended up changing the main solution from the first iteration of the PR. Let me try backing out this change.

Reply:
At least the release_assert needs to go away.  The redirect_post autest triggers it.  Once hit that, I think the safest thing is to mark ua_entry not in tunnel, so its contents get freed up during the HttpSM cleanup.

Reply:
OK, got it.